src/app/launcher/import-app/gitprovider-importapp-step/gitprovider-importapp-step.component.ts

```diff
@@ -200,9 +200,26 @@ export class GitproviderImportappStepComponent extends LauncherStep implements A
    * Update selection
    */
   updateGitHubSelection(): void {
+    if (this.launcherComponent.summary.gitHubDetails.repository) {
+      this.launcherComponent.summary.gitHubDetails.repositoryAvailable = true;
+    }
     this.initCompleted();
   }
 
+  /**
+   * Ensure repo name is available for the selected organization
+   */
+  getGitHubRepos(): void {
+    let org = this.launcherComponent.summary.gitHubDetails.organization;
+    this.launcherComponent.summary.gitHubDetails.repositoryList = [];
+    this.subscriptions.push(this.gitProviderService.getGitHubRepoList(org).subscribe((val) => {
+      if (val !== undefined) {
+        this.launcherComponent.summary.gitHubDetails.repositoryList = val;
+        this.initCompleted();
+      }
+    }));
+  }
```

Comment:
Selecting an organization will call getGitHubRepos multiple times; thus, we should destroy the previous subscription before creating a new one. Don't want multiple subscriptions that perform the same task -- this will help avoid potential memory leaks. 

Note that we still need a final unsubscribe in the ngOnDestroy, but getGitHubRepos would look more like this:

if (this.gitHubReposSubscription !== undefined) {
  this.gitHubReposSubscription.unsubscribe();
}
this.gitHubReposSubscription = this.gitProviderService.getGitHubRepoList(org).subscribe...

Reply:
Looking at the validateRepo method and it also appears to create multiple subscriptions. We should probably change that, too.

Alternatively, we could change the GitProviderService to update the Observable, when given a new org. We would then be able to subscribe only once in ngOnInit, for example. However, unsubscribing should work fine.

Reply:
Thanks Dan , have made the changes pls take  a look